src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1782,14 +1787,26 @@ private BaseListSyntax ParseBaseList(SyntaxToken typeKeyword, bool haveParameter
             {
                 _pool.Free(list);
             }
+
+            PostSkipAction skipBadBaseListTokens(ref SyntaxToken colon, SeparatedSyntaxListBuilder<BaseTypeSyntax> list, SyntaxKind expected)
+            {
+                return this.SkipBadSeparatedListTokensWithExpectedKind(ref colon, list,
+                    s_notExpectedForEndOfBaseList,
+                    s_abortForEndOfBaseList,
+                    expected);
+            }
         }
 
-        private PostSkipAction SkipBadBaseListTokens(ref SyntaxToken colon, SeparatedSyntaxListBuilder<BaseTypeSyntax> list, SyntaxKind expected)
+        private static readonly Func<LanguageParser, bool> s_notExpectedForEndOfBaseList = p => p.NotExpectedForEndOfBaseList();
+        private bool NotExpectedForEndOfBaseList()
```

Comment:
>NotExpectedForEndOfBaseList [](start = 21, length = 27)

It looks like there is only one call site, the extraction of the method feels unnecessary.